Lib/fontTools/ttLib/tables/ttProgram.py

```diff
@@ -248,9 +250,13 @@ def toXML(self, writer, ttFont):
 				writer.begintag("assembly")
 				writer.newline()
 				i = 0
+				indent = 0
 				nInstr = len(assembly)
 				while i < nInstr:
 					instr = assembly[i]
+					if _unindentRE.match(instr):
+						indent -= 1
+					writer.write("  " + "    " * indent)
```

Comment:
ah, since you are at it, could you also fix the inconsistency in the indentation levels for the glyph instructions?

```xml
<instructions><assembly>
    NPUSHB[ ]	/* 24 values pushed */
    26 72 28 16 17 175 28 1 38 28 17 17 28 38 3 3 20 74 13 3 33 74 3 9
    ...
    IUP[0]	/* InterpolateUntPts */
    IUP[1]	/* InterpolateUntPts */
  </assembly></instructions>
```

I think you just need to properly add a `newline()` before and after dumping the glyph program. Not sure it was not already like that:

```diff
diff --git a/Lib/fontTools/ttLib/tables/_g_l_y_f.py b/Lib/fontTools/ttLib/tables/_g_l_y_f.py
index a346813..437a646 100644
--- a/Lib/fontTools/ttLib/tables/_g_l_y_f.py
+++ b/Lib/fontTools/ttLib/tables/_g_l_y_f.py
@@ -375,7 +375,9 @@ class Glyph(object):
                                writer.newline()
                        if self.numberOfContours:
                                writer.begintag("instructions")
+                               writer.newline()
                                self.program.toXML(writer, ttFont)
+                               writer.newline()
                                writer.endtag("instructions")
                                writer.newline()
```